### PR TITLE
Fix test deprecations

### PR DESCRIPTION
**Pull request by benjaminwols**
Minitest has new syntax which showed a lot of deprecation warnings in the test suite.
This PR converts to the new syntax.

### CODE REVIEW

File: Rakefile [hunk 1]
```diff
@@ -4,6 +4,7 @@ require "rake/testtask"
 Rake::TestTask.new do |t|
   t.libs.push 'spec'
   t.test_files = FileList['spec/**/*_spec.rb']
+  t.warning = false
```
Comment (mvz) [inline Rakefile:7]: Any particular reason to turn of warnings here?

Comment (benjaminwols) [reply Rakefile:7]: No particular reason, I can revert the change if you want?

Comment (mvz) [reply Rakefile:7]: I prefer keeping warnings on, but I'll let @maartenvanvliet have the final say.

Comment (benjaminwols) [reply Rakefile]: I've reverted the change and merged newest master so tests are run

Comment (maartenvanvliet) [reply Rakefile]: Thx, yes I prefer keeping the warnings

**Comment by benjaminwols** [comment]
@maartenvanvliet I've updated the Readme badge to Github Actions and removed the deprecated code climate gem and updated Github actions to push coverage to code climate

File: Rakefile (hunk 1 above)
Comment (benjaminwols) [reply Rakefile]: done

**Review by maartenvanvliet — CHANGES_REQUESTED** [review]
Can't say I'm a fan of the `_(...).must_equal` syntax, and would probably prefer plain `assert_equal` assertions over this. But the library already uses specs and this is the new way so 🤷

Thanks for fixing it all up!

**Pull request merged**